src/mapping/RadialBasisFctSolver.hpp

```diff
@@ -467,6 +505,41 @@ Eigen::VectorXd RadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>::solveConsistent(E
   return out;
 }
 
+template <typename RADIAL_BASIS_FUNCTION_T>
+template <typename IndexConatiner>
+typename RadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>::DecompositionType RadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>::buildKernelDecomposition(const IndexConatiner &inputIds, double parameter) const
+{
+  if constexpr (RadiusInitialization<RADIAL_BASIS_FUNCTION_T>::isAvailable()) {
+    if constexpr (RadiusInitialization<RADIAL_BASIS_FUNCTION_T>::requiresConversion()) {
+      parameter = RadiusInitialization<RADIAL_BASIS_FUNCTION_T>::transformRadiusToShape(parameter);
+    }
+    RADIAL_BASIS_FUNCTION_T kernel(parameter);
+    fillKernelEntries(_kernelMatrix, kernel, _inputMesh, inputIds, _activeAxis);
+
+    if constexpr (RADIAL_BASIS_FUNCTION_T::isStrictlyPositiveDefinite()) {
+      return Eigen::LLT<Eigen::MatrixXd>(_kernelMatrix); // consider Eigen::LLT<Eigen::Ref<Eigen::MatrixXd>>
+    } else {
+      return Eigen::ColPivHouseholderQR<Eigen::MatrixXd>(_kernelMatrix); // consider Eigen::ColPivHouseholderQR<Eigen::Ref<Eigen::MatrixXd>>
+    }
+  }
+  PRECICE_UNREACHABLE("RBF does not support a radius-initialization and was still used to instantiate an optimizer.");
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+template <typename IndexConatiner>
+std::tuple<double, double> RadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>::computeErrorEstimate(const Eigen::VectorXd &inputData, const IndexConatiner &inputIds, double parameter) const
+{
+  _decMatrixC = buildKernelDecomposition(inputIds, parameter);
+
+  if (_decMatrixC.info() != Eigen::ComputationInfo::Success) {
+    return {std::numeric_limits<double>::infinity(), 0};
```

Comment:
returning infinity can be tricky as some compilers optimize infinity out.

If this is some kind of error condition, prefer to use a std:optional instead.